CONTRIBUTING.md

```diff
@@ -0,0 +1,23 @@
+## Releasing a new npm version
+
+- Change the version in `package.json`, make a new git tag, and push it to GitHub.
+- Wait until the GitHub Actions on the master branch pass.
+- The artifacts of the latest GitHub Action run should be downloaded from the actions tab of the GitHub repository
```

Comment:
This seems wrong. We should _not_ be downloading the artifacts of the latest GitHub action. We should be pulling the artifacts from the action that was built via the release tag that we pushed.

I would also prefer leveraging a different release system (I don't want to push artifacts for linux, osx, or 32 bit windows to the npm package). I am not familiar enough with prebuild enough here, but I vastly prefer node-pre-gyp, which pushes artifacts to s3 / github releases and downloads _only_ the artifacts necessary for the platform you're using. For instance, a Windows platform should never download linux/mac os binaries.

Reply:
Also, the downloading of artifacts is at install time with node-pre-gyp, rather than at publish time. I think that's where the difference is actually coming from.

Reply:
I don't recommend using anything other than `prebuildify` + `node-gyp-build`. I have tried all the possible solutions, and the most performant and reliable one is thing configuration. 

If you read the readme for prebuildify, the contributors of prebuild also recommend this over all the other solutions:

https://github.com/prebuild/prebuildify#prebuildify

With prebuildify, all prebuilt binaries are shipped inside the package that is published to npm, which means there's no need for a separate download step like you find in prebuild. The irony of this approach is that it is faster to download all prebuilt binaries for every platform when they are bundled than it is to download a single prebuilt binary as an install script.

Always use prebuildify --@mafintosh

Reply:
I understand that the prebuildify team suggests using prebuildify, but I've been involved with many packages that do not leverage prebuildify and they work wonders. I also like being able to _decouple_ the prebuild process, because this PR lacks the ability to precompile binaries for Electron - something that solutions like node-pre-gyp allow you to do easily. From my perspective a solution like nodegit leverages is much preferable.

Reply:
That's not quite accurate. The Electron artifacts are already downloaded. See this:
https://github.com/aminya/nsfw/suites/2602423618/artifacts/56919535

It's up to you, I have been using this configuration inside ZeroMQ, Atom, Nteract, etc.

Reply:
Mind, I think part of my point was missed as to why I don't like prebuildify's approach of generating all artifacts and publishing them together..  As someone who needs to package up and distribute an application for customers to download, every package that employs prebuildify is now a place where the install size of my application grows with no gain other than a slightly faster install time for developers. As that package grows in popularity, at least someone on my team has to put effort into tracking down which packages need additional pruning.

Reply:
I suppose if you definitely want to stay in the prebuild ecosystem, which is fine as it looks like it leverages GitHub storage easier than node-pre-gyp does, https://www.npmjs.com/package/prebuild-install is the more traditional way to get the binaries at install time rather than at publish time.

Reply:
We can use prebuild-install, but as already mentioned, it would be unreliable (downloading issues, rate limiting, etc) and the installation time would be more. But it seems that the "absolute" package size is more important than reliability and speed for you.

P.S: I am not supporting the 'prebuild' ecosystem just for the sake of it. We have been building a new all-in-one solution that fixes the issues of the current ecosystems, but it is still not stable.
https://github.com/EmbeddedEnterprises/cmake-ts

Reply:
>it would be unreliable (downloading issues, rate limiting, etc) and the installation time would be more.

Do you have a good write up from anyone illustrating what exactly is meant here by "unreliable" or "slow"? I think these words by themselves are pretty vague. How often are we seeing install failures due to unreliability in the community? How much slower is our install? If NSFW was installed as a module in an application like VSCode, Atom, or GitKraken - is the increase in time to install nsfw _the bottleneck_ that those teams will be looking to solve in their bootstrap procedures?

I would definitely say that without clear metrics to demonstrate how much more reliable/fast prebuildify is over node-pre-gyp or similar, it's not clear to me that we should leverage this process at the expense of introducing additional work for people who leverage this package. The way I am thinking about it is that VSCode, GitKraken, and Atom all ship a copy of NSFW, do we want them to be responsible for cleaning up our unused files, or do we want to make it our responsibility to ship a leaner package?

I am open to learning about these concerns, it could very well be a valuable thing to take back into NodeGit. I think I just need a little help answering these questions to see where my intuition is failing, and it's also possible that my intuition is on point, too.

I hope I'm not being rude. I really do appreciate the time and effort you put in here. I want to be clear where I'm failing to get behind prebuildify. I definitely need convincing to stay on the current course.

Reply:
Especially because I know this is a _highly requested feature_. :100:

Reply:
Well, I am not a fan of premature optimizations either. The installation size of NSFW is 18 MB. You could make that only 1MB if you merge this PR. This is 18X improvement. nodegit that you mentioned is over 100MB. You could make that to ~2MB if you provided prebuilds. 

![image](https://user-images.githubusercontent.com/16418197/116773555-8066ed00-aa1b-11eb-9a9e-15d74885cbee.png)
![image](https://user-images.githubusercontent.com/16418197/116773557-81981a00-aa1b-11eb-838b-51158910d9c7.png)


Also, as far as I know, prebuildify is the only solution that is able to provide prebuilds for Napi + Electron without requiring any rebuilding afterward. This is huge for me and the users of Atom, Ntract, ZeroMQ, etc. 

The prebuild organization still recommends prebuildify over prebuild/prebuild-install. 
https://github.com/prebuild/prebuild-install/pull/150

Reply:
Your numbers are straight up wrong.

For one, you've dropped the `pdb` files which are a _requirement_ to send out with any prebuilts (necessary to setup crash reporting in applications). When including the pdb file for every single `.node` on windows you're going to see roughly a 8 meg increase per `.pdb` file, which there will be a 1-to-1 relationship with the windows `.node` files.

Your _absolute baseline_ if you're shipping the correct prebuilts is:
1 napi build per OS at roughly: .5 megs
an additional napi build per OS for electron base each also roughly .5 megs
`pdb` files for all 4 windows builds 8 megs a piece

For a grand total of 36 megs, twice the install size you're listing for nsfw. No thank you.

Reply:
Also, neat side-effect of this I discovered in nodegit we were missing a clean operation on our electron prebuilder that we're running for the node releases.